LambdaS3FileZipper.IntegrationTests/Aws/EnvironmentVariableS3TestEnvironment.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using LambdaS3FileZipper.IntegrationTests.Aws.Interfaces;
+
+namespace LambdaS3FileZipper.IntegrationTests.Aws
+{
+	public class EnvironmentVariableS3TestEnvironment : IS3TestEnvironment
+	{
+		private const string TestBucketEnvironmentVariable = "S3TestBucket";
+		private const string TestObjectEnvironmentVariable = "S3TestObject";
+
+		private string testBucket;
+
+		public string TestBucket
+		{
+			get
+			{
+				if (testBucket == null)
+				{
+					testBucket = Environment.GetEnvironmentVariable(TestBucketEnvironmentVariable);
+
+					if (testBucket == null)
+					{
+						throw new Exception($"missing required environment variable {TestBucketEnvironmentVariable}");
+					}
+				}
+
+				return testBucket;
+			}
+		}
```

Comment:
**Nit:** can we simplify this bit:

```
=> testBucket
    ?? Environment.GetEnvironmentVariable(TestBucketEnvironmentVariable)
    ?? throw new Exception($"missing required environment variable {TestBucketEnvironmentVariable}");
```

Reply:
That's OK: this is a micro-optimization as I see it 😏 . I'd remove the local field altogether, but still just a **nit**.